### DISCUSSION TITLE
SUNTF objects

### DISCUSSION THREAD

**Discussion by huansouza** [FeynCalc/feyncalc#131] [General]
Hi everyone,

I'm having some trouble understanding SUNTF objects. In my computation I have the following output:
![Screenshot_20210729_153830](https://user-images.githubusercontent.com/62699121/127547864-c130d945-2150-4fbe-af6a-b1fbcdd19dd6.png)

My final result should be zero, but I'm not sure if the above result is.

Thank you all!

**Comment by vsht** [comment]
Hi,

a `SUNTF` is just a color matrix with explicit fundamental indices, nothing more than that. There can be particular
simplifications (i.e. color Fierz identities) that are not implemented yet, so when working when `SUNTF`s on the amplitude
level, one often has to do things by hand. When the amplitude is squared, products of  `SUNTF`s are automatically turned into 
color traces.

Anyway, your output looks weird. How can it be that the first two terms have fundamental color indices Col3,Col4 and the last two terms
Col4,Col5? 

The result is proportional to the difference of two Kronecker deltas, cf.

```
ruleSUNTFSplit = SUNTF[{a_SUNIndex, b_SUNIndex},
   i1_SUNFIndex, i2_SUNFIndex] :> (ui = Unique[]; 
   SUNTF[{SUNIndex[a]}, i1, SUNFIndex[ui]] SUNTF[{SUNIndex[b]}, 
     SUNFIndex[ui], i2])

ruleFierz = 
 SUNTF[{a_SUNIndex}, i_SUNFIndex, j_SUNFIndex] SUNTF[{b_SUNIndex}, 
    k_SUNFIndex, l_SUNFIndex] :>
  
  1/2 (SUNFDelta[i, l] SUNFDelta[k, j] - 
     1/SUNN SUNFDelta[i, j] SUNFDelta[k, l])

SUNTF[{Glu1, Glu2}, Col4, Col3] + SUNTF[{Glu2, Glu1}, Col4, Col3] - 
 SUNTF[{Glu1, Glu2}, Col4, Col5] - SUNTF[{Glu2, Glu1}, Col5, Col4]
FCI[%] /. ruleSUNTFSplit /. ruleFierz // SUNSimplify
(*2*CF*(SUNFDelta[SUNFIndex[Col3], SUNFIndex[Col4]] - SUNFDelta[SUNFIndex[Col4], SUNFIndex[Col5]])*)
```
